src/main/java/com/zendesk/maxwell/row/RowMapDeserializer.java

```diff
@@ -99,12 +105,32 @@ public RowMap deserialize(JsonParser jsonParser, DeserializationContext deserial
 		return rowMap;
 	}
 
-	private Object getValue(ValueNode value)
+	@VisibleForTesting
+	final Object getValue(ValueNode value)
 	{
 		if (value.isNull()) {
 			return null;
 		}
 
+		if (value.numberType() != null) {
+			switch (value.numberType()) {
+				case LONG:
+					return value.longValue();
+				case DOUBLE:
+					return value.doubleValue();
+				case FLOAT:
+					return value.floatValue();
+				case INT:
+					return value.intValue();
+				case BIG_DECIMAL:
+					return value.decimalValue();
+				case BIG_INTEGER:
+					return value.bigIntegerValue();
+				default:
+					return value.asText();
+			}
+		}
+
```

Comment:
You can remove the if `value.canConvertToInt()` and `value.canConvertToLong()` blocks below.